accounts/forms.py

```diff
@@ -24,18 +24,21 @@
 from django.utils.translation import ugettext as _
 from django.utils.safestring import mark_safe
 from django.core.urlresolvers import reverse
+from multiupload.fields import MultiFileField
 from accounts.models import Profile
 from utils.forms import RecaptchaForm, HtmlCleaningCharField, filename_has_valid_extension
 from utils.spam import is_spam
 
 
-def validate_file_extension(value):
-    if not filename_has_valid_extension(str(value)):
-        raise forms.ValidationError('Uploaded file format not supported or not an audio file.')
+def validate_file_extension(audiofiles):
+    for file_ in audiofiles:
+        content_type = file_.content_type
+        if not content_type.startswith("audio") and not filename_has_valid_extension(str(file_)):
```

Comment:
The one I mean is the second one you propose, and you're right that is a bit confusing (actually I just realized it is wrong).